airbyte-webapp/src/components/ui/Toast/Toast.tsx

```diff
@@ -1,86 +1,64 @@
-import { faTimes } from "@fortawesome/free-solid-svg-icons";
+import { faCheck, faExclamation, faTimes } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import classNames from "classnames";
 import React from "react";
-import styled, { keyframes } from "styled-components";
-
-import { H5 } from "components/base/Titles";
 
 import { Button } from "../Button";
-import { ErrorSign } from "./ErrorSign";
+// eslint-disable-next-line css-modules/no-unused-class
 import styles from "./Toast.module.scss";
 
+export enum ToastType {
+  WARNING = "warning",
+  SUCCESS = "success",
+  ERROR = "error",
+  INFO = "info",
+}
+
 interface ToastProps {
   title: string | React.ReactNode;
   text?: string | React.ReactNode;
+  type?: ToastType;
   hasError?: boolean;
+  onAction?: () => void;
   onClose?: () => void;
 }
 
-export const SlideUpAnimation = keyframes`
-  0% {
-    translate(-50%, -100%);
-    bottom: -49px;
-  }
-  100% {
-    translate(-50%, 0);
-    bottom: 49px;
-  }
-`;
-
-const Singleton = styled.div<{ hasError?: boolean }>`
-  position: fixed;
-  bottom: 49px;
-  left: 50%;
-  transform: translate(-50%, 0);
-  z-index: 20;
-
-  padding: 25px 25px 22px;
-
-  background: ${({ theme, hasError }) => (hasError ? theme.lightDangerColor : theme.lightPrimaryColor)};
-  border: 1px solid ${({ theme }) => theme.greyColor20};
-  box-shadow: 0 1px 2px ${({ theme }) => theme.shadowColor};
-  border-radius: 8px;
+const ICON_MAPPING = {
+  [ToastType.WARNING]: faExclamation,
+  [ToastType.ERROR]: faTimes,
+  [ToastType.SUCCESS]: faCheck,
+  [ToastType.INFO]: faExclamation,
+};
 
-  display: flex;
-  flex-direction: row;
-  align-items: center;
-
-  animation: ${SlideUpAnimation} 0.25s linear;
-`;
-
-const Title = styled(H5)<{ hasError?: boolean }>`
-  color: ${({ theme, hasError }) => (hasError ? theme.dangerColor : theme.primaryColor)};
-
-  font-style: normal;
-  font-weight: bold;
-  font-size: 15px;
-  line-height: 18px;
-`;
-
-const Text = styled.div`
-  color: ${({ theme }) => theme.mediumPrimaryColor};
+function getIcon(toastType: ToastType) {
+  return ICON_MAPPING[toastType];
+}
```

Comment:
This function seems unnecessary?

Reply:
I am using this function to show the icon based on the type of error. Should I default to ! icon then?